activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -200,70 +233,98 @@ def setup
         "test"        => { "database" => "test-db" },
         "production"  => { "url" => "prod-db-url" }
       }
-
-      ActiveRecord::Base.stubs(:configurations).returns(@configurations)
-      ActiveRecord::Base.stubs(:establish_connection)
     end
 
     def test_creates_current_environment_database
-      assert_called_with(
-        ActiveRecord::Tasks::DatabaseTasks,
-        :create,
-        ["database" => "test-db"],
-      ) do
-        ActiveRecord::Tasks::DatabaseTasks.create_current(
-          ActiveSupport::StringInquirer.new("test")
-        )
+      with_stubbed_configurations_establish_connection do
+        assert_called_with(
+          ActiveRecord::Tasks::DatabaseTasks,
+          :create,
+          ["database" => "test-db"],
+        ) do
+          ActiveRecord::Tasks::DatabaseTasks.create_current(
+            ActiveSupport::StringInquirer.new("test")
+          )
+        end
       end
     end
 
     def test_creates_current_environment_database_with_url
-      assert_called_with(
-        ActiveRecord::Tasks::DatabaseTasks,
-        :create,
-        ["url" => "prod-db-url"],
-      ) do
-        ActiveRecord::Tasks::DatabaseTasks.create_current(
-          ActiveSupport::StringInquirer.new("production")
-        )
+      with_stubbed_configurations_establish_connection do
+        assert_called_with(
+          ActiveRecord::Tasks::DatabaseTasks,
+          :create,
+          ["url" => "prod-db-url"],
+        ) do
+          ActiveRecord::Tasks::DatabaseTasks.create_current(
+            ActiveSupport::StringInquirer.new("production")
+          )
+        end
       end
     end
 
     def test_creates_test_and_development_databases_when_env_was_not_specified
-      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
-        with("database" => "dev-db")
-      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
-        with("database" => "test-db")
-
-      ActiveRecord::Tasks::DatabaseTasks.create_current(
-        ActiveSupport::StringInquirer.new("development")
-      )
+      with_stubbed_configurations_establish_connection do
+        assert_called_with(
+          ActiveRecord::Tasks::DatabaseTasks,
+          :create,
+          [
+            ["database" => "dev-db"],
+            ["database" => "test-db"]
+          ],
+          returns: true
```

Comment:
This return value seems redundant

Reply:
Or may be we should add this for all the kind of tests in order to ensure that dbs were created?

Reply:
I am playing with it and wonder why changes `returns: false` doesn't cause the test failure

Reply:
By looking at the implementation of the `create` method looks like we shouldn't expect returning any value.

Reply:
Agreed.